### ISSUE TITLE
transpose(l, fill=NULL)

### ISSUE THREAD

**Issue by stla** [Rdatatable/data.table#2975]
Hello, 

 In `transpose.c`:
```
  if (length(fill) != 1)
    error("fill must be NULL or length=1 vector.");
```
Hence `transpose(l, fill=NULL)` returns this error message, because `length(NULL)` is `0`. So it's not true that `fill` can be `NULL`. Perhaps you meant `NA` ?

**Comment by MichaelChirico** [comment]
@stla yes, this is documented in `?transpose`. I'll change the error message.
